frontend/store/user.js

```diff
@@ -0,0 +1,79 @@
+import moment from 'moment'
+
+export const state = () => ({
+  id: null,
+  currentlyAllowedProjectId: -1,
+  restingUserId: null,
+  restingEndTime: null
+})
+
+export const mutations = {
+  setUserId(state, userId) {
+    state.id = userId
+  },
+  setCurrentlyAllowedProjectId(state, projectId) {
+    state.currentlyAllowedProjectId = projectId
+  },
+  setRestingPeriod(state, endTime) {
+    state.restingUserId = state.id
+    state.restingEndTime = endTime
+  },
+  clearRestingPeriod(state) {
+    state.restingUserId = null
+    state.restingEndTime = null
+  }
+}
+
+export const getters = {
+  getUserId(state) {
+    return state.id
+  },
+  getCurrentlyAllowedProjectId(state) {
+    return state.currentlyAllowedProjectId
+  },
+  getRestingUserId(state) {
+    return state.restingUserId
+  },
+  getRestingEndTime(state) {
+    if (state.restingEndTime !== null) {
+      return moment(state.restingEndTime, 'ddd, DD-MM-YYYY HH:mm:ss').toDate()
+    }
+    return null
+  }
+}
+
+export const actions = {
+  setCurrentlyAllowedProjectId({ commit }, projectId) {
+    commit('setCurrentlyAllowedProjectId', projectId)
+  },
+  setRestingPeriod({ commit }) {
+    const startTime = new Date()
+    const endTime = moment(startTime).add(5, 'm').format('ddd, DD-MM-YYYY HH:mm:ss')
+    commit('setRestingPeriod', endTime)
+  },
+  getRestingPeriod({ commit, getters }) {
+    const currentTime = new Date()
+    const restingEndTime = getters.getRestingEndTime
+
+    if (currentTime >= restingEndTime) {
+      commit('clearRestingPeriod')
```

Comment:
I dont remember if I missed this previously, but I think a getter should only do one think, to `get` the data, and not to clear the data :"   how about separating this to a new function?

Reply:
Hello, this one is actually not a "getter", it is actually an action. I need the action to wrap the real getter (that one on the top with `getters.getRestingEndTime` together with mechanism to clear the resting time data when the resting period has finished. Notably, the real getter (`getters.getRestingEndTime`) only does one thing which is getting the data. Maybe I should rename this action from `getRestingPeriod` to something like `checkRestingPeriod` or `calculateRestingPeriod`?

Reply:
Yes, I think that will be more clear XD sorry for the long reply

Reply:
Or probably you can separate xD so we can have both functions, get... and clear... Depending on your preference.